app/services/api/v1x1/catalog/discard_restore.rb

```diff
@@ -0,0 +1,51 @@
+module Api
+  module V1x1
+    module Catalog
+      module DiscardRestore
+        extend ActiveSupport::Concern
+
+        CHILD_DISCARD_TIME_LIMIT = 30
+
+        module ClassMethods
+          def destroy_dependencies(*klass_names)
+            klass_names.each do |klass|
+              define_method "discard_#{klass}" do
```

Comment:
This file was added here as this line was blowing up in `app/services/api/v1x1/catalog.rb` about an invalid superclass.  I troubleshooted for a while but then gave up and added a version here instead.

Reply:
I wonder if it's because it's a module and not a class. Not sure how that works when you try to nest and declare both in the same file. I think solving that potential "issue" could definitely be pushed to another PR though.